web/modules/custom/custom_weather/custom_weather.install

```diff
@@ -0,0 +1,54 @@
+<?php
+
+use Drupal\Core\Database\Database;
+
+/**
+ * Implements hook_schema().
+ */
+function custom_weather_schema(): array {
+  $schema = [];
+
+  $schema['custom_weather_data'] = [
+    'description' => 'Stores user-selected cities for weather data.',
+    'fields' => [
+      'id' => [
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+        'description' => 'Primary key for the custom weather data table.',
+      ],
+      'user_id' => [
+        'type' => 'int',
+        'size' => 'normal',
+        'not null' => TRUE,
+        'unsigned' => TRUE,
+        'description' => 'User ID from the users table.',
+      ],
+      'city' => [
+        'type' => 'varchar',
+        'length' => 50,
+        'not null' => TRUE,
+        'description' => 'Selected city name.',
+      ],
+    ],
+    'primary key' => ['id'],
+  ];
+
+  return $schema;
+}
+
+/**
+ * Implements hook_update_N().
+ */
+function custom_weather_update_10001(): void {
+  $schema = custom_weather_schema();
+  $table = $schema['custom_weather_data'];
+
+  // Check if the table already exists.
+  if (Database::getConnection()->schema()->tableExists('custom_weather_data')) {
```

Comment:
More reliable to use database service to get connections, instead of static method on the Database class